src/ansys/simai/core/data/models.py

```diff
@@ -20,23 +20,238 @@
+
+            # Get the last configuration from a project
+            bld_conf = my_project.last_model_configuration
+
+            # Define a new axis for the Domain of Analysis
+            new_doa_height = DomainAxisDefinition("relative_to_min", 180.5, 99.1)
+
+            # Assign the new Domain of Analysis to the configuration
+            bld_conf.domain_of_analysis.height = new_doa_height
+
+    """
+
+    def __post_init__(self):
+        """Assess whether the value and length are correctly set."""
+
+        self.__validate_length(self.length)
+        self.__validate_value(self.value)
+
+    def __set_value(self, val: float):
+        self.__validate_value(val)
+        self.__dict__["value"] = val
+
+    def __get_value(self):
+        return self.__dict__.get("value")
+
+    def __validate_value(self, val: float):
+        if val < 0 and self.position != "absolute":
+            raise InvalidArguments(
+                f"{self.__class__.__name__}: 'value' must be a positive number when the position is not 'absolute'.",
+            ) from None
+
+    def __set_length(self, lgth: float):
+        self.__validate_length(lgth)
+        self.__dict__["length"] = lgth
+
+    def __get_length(self):
+        return self.__dict__.get("length")
+
+    def __validate_length(self, lgth: float):
+        if not lgth > 0:
+            raise InvalidArguments(
+                f"{self.__class__.__name__}: 'length' must be a positive number.",
+            ) from None
+
+    position: Literal["relative_to_min", "relative_to_max", "relative_to_center", "absolute"]
+    value: float = property(__get_value, __set_value)
+    length: float = property(__get_length, __set_length)
+
+    del __set_length, __set_value, __get_length, __get_value
+
+
+@dataclass
+class DomainOfAnalysis:
+    """Defines the Domain of Analysis.
+
+    Args:
+        length: Domain of Analysis along the X axis
+        width: Domain of Analysis along the Y axis
+        height: Domain of Analysis along the Z axis
+
+    Example:
+        Get the Domain of Analysis from a configuration and replace it with a new one
+
+        .. code-block:: python
+
+            from ansys.simai.core.data.models import DomainOfAnalysis
+
+            # Get the last configuration from a project
+            bld_conf = my_project.last_model_configuration
+
+            # Define a new Domain of Analysis
+            new_doa = DomainOfAnalysis(
+                length=("relative_to_max", 5, 8.1),
+                width=("relative_to_max", 5, 8.1),
+                height=("absolute", -4.5, 0.1),
+            )
+
+            # Assign the new Domain of Analysis to the configuration
+            bld_conf.domain_of_analysis = new_doa
+
+    """
+
+    length: DomainAxisDefinition = None
+    width: DomainAxisDefinition = None
+    height: DomainAxisDefinition = None
+
+    def __post_init__(self):
+        if type(self.length) == tuple:
```

Comment:
```suggestion
        if isinstance(self.length, tuple):
```
To do everywhere, `type()` does not take inheritance into account